integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala

```diff
@@ -179,6 +193,68 @@ object CommonUtil {
     isValid
   }
 
+  def validateTypeConvert(desType: Option[String], value: String): Boolean = {
+    var flag = true
+    val result = desType match {
+      case Some("IntegerType") =>
+        DataTypeUtil.getDataBasedOnDataType(value.trim, DataType.INT)
+      case Some("StringType") =>
+        DataTypeUtil.getDataBasedOnDataType(value.trim, DataType.STRING)
+      case Some("LongType") =>
+        DataTypeUtil.getDataBasedOnDataType(value.trim, DataType.LONG)
+      case Some("FloatType") =>
+        DataTypeUtil.getDataBasedOnDataType(value.trim, DataType.FLOAT)
+      case Some("DoubleType") =>
+        DataTypeUtil.getDataBasedOnDataType(value.trim, DataType.DOUBLE)
+      case Some("ByteType") =>
+        DataTypeUtil.getDataBasedOnDataType(value.trim, DataType.BYTE)
+      case Some("ShortType") =>
+        DataTypeUtil.getDataBasedOnDataType(value.trim, DataType.SHORT)
+      case Some("BooleanType") =>
+        DataTypeUtil.getDataBasedOnDataType(value.trim, DataType.BOOLEAN)
+      case Some("DecimalType") =>
```

Comment:
DecimalType comes with precision and scale like DecimalType(10,0)...so validation for decimal type will fail here....Please correct the validation logic for DecimalType